src/components/Airdrop/components/Airdrop/Airdrop.tsx

```diff
@@ -9,34 +8,32 @@ import { t } from "i18next";
 import { useCallback, useEffect, useState } from "react";
 import { useSelector } from "react-redux";
 import { RootState } from "reducers";
-import { isEthereumProvider, normalizeAddress } from "../../../../utils";
-import TokenAirdrop from "../TokenAirdrop/TokenAirdrop";
+import { normalizeAddress } from "../../../../utils";
+//import TokenAirdrop from "../TokenAirdrop/TokenAirdrop";
 import NFTAirdrop from "../NFTAirdop/NFTAirdrop";
+import { useParams } from "react-router-dom";
+import { useActions } from "actions/contractsActions";
 import "./index.scss";
-import { useSearchParams } from "react-router-dom";
 
 export default function Airdrop() {
-  const [userInput, setUserInput] = useState("");
-  const [pendingWallet, setPendingWallet] = useState(false);
+  const { hasClaimed, isRedeemed } = useActions();
+  const { walletAddress } = useParams();
+  const [userInput, setUserInput] = useState(walletAddress);
   const nftET = useSelector((state: RootState) => state.dataReducer.airdrop?.nft);
   const tokenET = useSelector((state: RootState) => state.dataReducer.airdrop?.token);
   const [nftEligibilityStatus, setNFTEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
   const [tokenId, setTokenId] = useState<string>();
-  const [tokenEligibilityStatus, setTokenEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
-  const [tokenAmount, setTokenAmount] = useState<number>();
+  //const [tokenEligibilityStatus, setTokenEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
+  //const [tokenAmount, setTokenAmount] = useState<number>();
   const [inTokenAirdop, setInTokenAirdrop] = useState(false);
-  const [searchParams] = useSearchParams();
-  const walletAddress = searchParams.get("walletAddress");
 
-
-  const handleChange = useCallback(async (input: string) => {
-    setUserInput(input);
-    if (isAddress(input)) {
-      const address = normalizeAddress(input);
-      if (Object.values(nftET!).includes(address)) {
-        const tokenId = Object.keys(nftET!).find(key => nftET![key] === address);
-        setTokenId(tokenId!);
-        if (await isRedeemed(tokenId!, address)) {
+  const checkAirdrop = useCallback(async () => {
+    if (userInput && isAddress(userInput) && nftET) {
+      const address = normalizeAddress(userInput);
+      if (Object.values(nftET).includes(address)) {
+        const tokenId = Object.keys(nftET!).find(key => nftET[key] === address);
+        setTokenId(tokenId);
+        if (tokenId && await isRedeemed(tokenId, address)) {
```

Comment:
its better to read the contract as a hook as in useCall in DepositWithdraw, the actual definition of the contract is defined in contract hooks file